src/classes/ADDR_Contact_TDTM.cls

```diff
@@ -135,13 +147,13 @@ public class ADDR_Contact_TDTM extends TDTM_Runnable {
             	//     - if MailingAddress field changes, update/new address object
                 //     - if isAddressOverride changes to false, refill from Default Address
             	
-            	Contact conOld = (Contact)listOld[i];
-            	
             	// new highest priority! if Account changes... 
-                if (conOld != null && con.AccountId != conOld.AccountId) {
+                if (acctChanged) {
                     // if no override, refill from the Default Address
             	    if (!con.is_Address_Override__c && !isContactAddressChanged(con, conOld)) {
-                        listConAddrReset.add(con);
+                        if (acctHasAddress) { // refresh address if the contact does not have one and account does
```

Comment:
I think that was what I originally did but then changed it to create on line 204 when I thought the contacts address should be added to an account the same as a new contact would. Let me take another look at this.